packages/transaction-controller/src/TransactionController.ts

```diff
@@ -3786,6 +3794,20 @@ export class TransactionController extends BaseController<
     this.#updateTransactionInternal(
       { transactionId, skipHistory: true },
       (txMeta) => {
+        const userFeeLevel = txMeta.userFeeLevel as GasFeeEstimateLevel;
+
+        if (Object.values(GasFeeEstimateLevel).includes(userFeeLevel)) {
+          txMeta.txParams.maxFeePerGas =
+            (gasFeeEstimates as FeeMarketGasFeeEstimates)?.[userFeeLevel]
```

Comment:
Rather than trying every property, could we do it based on the `type` in `gasFeeEstimates` which uses the `GasFeeEstimateType` type?